projects/postgresql.org/package.yml

```diff
@@ -0,0 +1,82 @@
+distributable:
+   url: https://github.com/postgres/postgres/archive/refs/tags/REL_{{version.major}}_{{version.minor}}.tar.gz
+
+   strip-components: 1
+versions:
+  - 15.2 # need to hardcode this version, because no easy way to get the tag yet
+
+dependencies:
+  openssl.org: ^1.0.1
+  gnu.org/readline: '*'
+  zlib.net: '*'
+  lz4.org: '*'
+  gnome.org/libxml2: '*'
+  gnome.org/libxslt: '*'
+
+build:
+  dependencies:
+    tea.xyz/gx/make: '*'
+    freedesktop.org/pkg-config: '*'
+    darwin:
+      tea.xyz/gx/cc: c99
+    linux:
+      gnu.org/gcc: '*'
+      gnu.org/bison: '*'
+      github.com/westes/flex: ^2.5.31
+      perl.org: '*'
+  script: |
+    ./configure $ARGS
+    make --jobs {{ hw.concurrency }}
+    make install
+  env:
+    ARGS:
+      - --prefix={{prefix}}
+      - --with-ssl=openssl
+      - --with-lz4
+      - --with-libxml
+      - --with-libxslt
+
+provides:
+  - bin/clusterdb
+  - bin/createdb
+  - bin/dropdb
+  - bin/dropuser
+  - bin/ecpg
+  - bin/initdb
+  - bin/pg_amcheck
+  - bin/pg_archivecleanup
+  - bin/pg_basebackup
+  - bin/pg_config
+  - bin/pg_controldata
+  - bin/pg_ctl
+  - bin/pg_dump
+  - bin/pg_dumpall
+  - bin/pg_isready
+  - bin/pg_receivewal
+  - bin/pg_recvlogical
+  - bin/pg_resetwal
+  - bin/pg_restore
+  - bin/pg_rewind
+  - bin/pg_test_fsync
+  - bin/pg_test_timing
+  - bin/pg_upgrade
+  - bin/pg_verifybackup
+  - bin/pg_waldump
+  - bin/pgbench
+  - bin/postgres
+  - bin/psql
+  - bin/reindexdb
+  - bin/vacuumdb
+
+test:
+  script: |
+    postgres -V
+
+    # this won't work because it requires 'su' priviledges
+    # mkdir -p ./data
+    # initdb -D ./data
+    # pg_ctl -D ./data -l logfile start
+    # createdb test
+    # psql -c 'create table test (id int);' test
+    # dropdb test
+    # pg_ctl -D ./data stop
```

Comment:
This actually works in darwin, but fails in linux. Not sure how I can write a better test that doesn't requires sudo priviledges.

Reply:
The error locally is 

```
initdb: error: cannot be run as root
initdb: hint: Please log in (using, e.g., "su") as the (unprivileged) user that will own the server process.
```

Reply:
Oh, wow. It won't run as root. Hm. Does `su - nobody -c 'initdb -D ./data'` work?

Reply:
If you need to, you could write the script as a fixture, and just do `sudo -u nobody sh $FIXTURE`

Reply:
I tried that as well. 

```
+ su - postgres -c 'initdb -D ./data'
-bash: initdb: command not found
```

Reply:
sudo -p preserves environment, so `SUDO="sudo -p -u nobody"`, maybe.

Reply:
All really annoying, though.

Reply:
Will look at it later again. sometimes it helps to step away from keyboard :)

Reply:
s_sometimes_pretty much always_